main.go

```diff
@@ -23,17 +25,35 @@ func printUsage() {
 	os.Exit(0)
 }
 
+func timeoutSecondsFromCount(count int) time.Duration {
+	waitOnLastOne := time.Duration(2) * time.Second
+	return time.Duration(count)*time.Second + waitOnLastOne
+
+}
+
 // RunPinger takes an Option object, gets a Dialer, and runs a Pinger against
 // the passed target, for count packets.
 func RunPinger(opt *vpn.Options, target string, count uint32) error {
-	conn := vpn.NewRawDialer(opt)
-	pinger := extras.NewPinger(conn, target, int(count))
-	err := pinger.Run()
+
+	c := int(count)
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutSecondsFromCount(c)+extraTimeoutSeconds)
+	defer cancel()
+
+	rawDialer := vpn.NewRawDialerWithContext(opt, ctx)
```

Comment:
The idiomatic usage is to always pass the `ctx` as the first argument to any function (and, FWIW, they also recommend not ever storing it inside structs but always passing it as argument).

You can read more about these ~rules here: https://pkg.go.dev/context

Based on these rules, I would say the idiomatic usage here would be to have `rawDialer.DialContext(ctx)`